docs/architecture-decision-record/ADR-0008 Integration events.md

```diff
@@ -0,0 +1,183 @@
+# ADR-0008 Integration events with protocol buffers
+
+* Status: accepted
+* Deciders: @sondergaard, @MartinFHansen
+* Date: 2021-08-23
+
+This document describes how integration events are defined, distributed and consumed in the system architecture of GreenEnergyHub.
+
+## Defining integration events
+
+All integration events are defined as [protocol buffer contracts](https://developers.google.com/protocol-buffers "Protocol Buffers | Google Developers"). By using this message technology, we get some benefits.
+
+* Strong contracts
+* Supports new fields fields to existing contracts
+* Tooling that supports the programming languages used in GreenEnergyHub
+* Fast and efficient (de)serializing
+* Binary format that is fast to transmit over the wire
+
+### Designing contracts
+
+When a contract is made public available it's considered final. This means that all modifications to the contract is backwards compatible. You are allowed to add new fields to a contract. Ordering of fields, data types, removal of fields etc. are considered breaking changes. If a breaking change is unavoidable then the contract should be duplicated and given a new name.
```

Comment:
```suggestion
When a contract is made public available it's considered final. This means that all future modifications to the contract must be backwards compatible. You are allowed to add new fields to a contract. Ordering of fields, data types, removal of fields etc. are considered breaking changes. If a breaking change is unavoidable then the contract must be duplicated and given a new name.
```

Reply:
Can you provide a guideline on the naming convension of duplicated contracts - e.g. <existing-contract-name>-v2?